src/fsharp/FSharp.Core/Linq.fs

```diff
@@ -87,19 +155,21 @@ module LeafExpressionConverter =
                         if isg1 then minfo2.IsGenericMethod && gmd = minfo2.GetGenericMethodDefinition()
                         else minfo = minfo2
                      ) ->
-                Some (obj, (minfo2.GetGenericArguments() |> Array.toList), args)
+                Some (obj, minfo2, args)
             | _ -> None)
 
-
     let (|SpecificCallToMethod|_|) (mhandle: System.RuntimeMethodHandle) =
         let minfo = (System.Reflection.MethodInfo.GetMethodFromHandle mhandle) :?> MethodInfo
         SpecificCallToMethodInfo minfo
+        
+    let (|SpecificCallToGenericMethod|_|) (mhandle: System.RuntimeMethodHandle) =
+        (|SpecificCallToMethod|_|) mhandle >> Option.map (fun (obj, minfo, args) -> obj, minfo.GetGenericArguments() |> Array.toList, args)
```

Comment:
Please remove the use of `>>` in FSharp.Core and make the lambda explicit (it makes for better debugging)